lib/core/include/experimental_plugin_framework.hpp

```diff
@@ -85,7 +85,7 @@ namespace irods::experimental::api {
                 THROW(err, "failed to perform the invocation");
             }
 
-            return json::parse({(char*)resp->buf, (char*)resp->buf+resp->len});
+            return json::parse((char*)resp->buf, (char*)resp->buf+resp->len);
```

Comment:
Why remove the braces? With the braces, (1) should be selected, but I'm a little confused as to how it builds without them.
```c++
// (1)
template<typename InputType>
static basic_json parse(InputType&& i,
                        const parser_callback_t cb = nullptr,
                        const bool allow_exceptions = true,
                        const bool ignore_comments = false);

// (2)
template<typename IteratorType>
static basic_json parse(IteratorType first, IteratorType last,
                        const parser_callback_t cb = nullptr,
                        const bool allow_exceptions = true,
                        const bool ignore_comments = false);

```
https://json.nlohmann.me/api/basic_json/parse/

Reply:
Ooh, secret (3). Okay, good to know.

Reply:
Yeah, just tried compiling with the braces, and it is indeed trying to use the deprecated version. 

```
In file included from irods/server/core/src/rodsAgent.cpp:33:
In file included from irods/server/core/include/plugin_lifetime_manager.hpp:4:
irods/lib/core/include/experimental_plugin_framework.hpp:88:26: warning: 'parse' is deprecated: Since 3.8.0; use parse(ptr, ptr + len) [-Wdeprecated-declarations]
            return json::parse({(char*)resp->buf, (char*)resp->buf+resp->len});
                         ^
irods-externals/json3.10.4-0/include/nlohmann/json.hpp:6947:5: note: 'parse' has been explicitly marked deprecated here
    JSON_HEDLEY_DEPRECATED_FOR(3.8.0, parse(ptr, ptr + len))
    ^
irods-externals/json3.10.4-0/include/nlohmann/thirdparty/hedley/hedley.hpp:1043:75: note: expanded from macro 'JSON_HEDLEY_DEPRECATED_FOR'
    #define JSON_HEDLEY_DEPRECATED_FOR(since, replacement) __attribute__((__deprecated__("Since " #since "; use " #replacement)))
                                                                          ^
```
Additionally, it fails when trying to construct a `detail::span_input_adapter`:
```
In file included from irods/server/core/src/rodsAgent.cpp:3:
In file included from irods/server/core/include/rsApiHandler.hpp:13:
In file included from irods/lib/core/include/apiHandler.hpp:7:
In file included from irods/lib/core/include/irods_plugin_base.hpp:12:
In file included from irods/server/re/include/irods_re_namespaceshelper.hpp:4:
In file included from irods/lib/core/include/irods_server_properties.hpp:7:
In file included from irods/lib/core/include/irods_configuration_parser.hpp:19:
In file included from irods-externals/json3.10.4-0/include/nlohmann/json.hpp:57:
In file included from irods-externals/json3.10.4-0/include/nlohmann/detail/input/binary_reader.hpp:17:
irods-externals/json3.10.4-0/include/nlohmann/detail/input/input_adapters.hpp:472:11: error: no matching constructor for initialization of 'nlohmann::detail::contiguous_bytes_input_adapter' (aka 'iterator_input_adapter<const char *>')
        : ia(input_adapter(first, last)) {}
          ^  ~~~~~~~~~~~~~~~~~~~~~~~~~~
irods/lib/core/include/experimental_plugin_framework.hpp:88:32: note: in instantiation of function template specialization 'nlohmann::detail::span_input_adapter::span_input_adapter<char *, 0>' requested here
            return json::parse({(char*)resp->buf, (char*)resp->buf+resp->len});
                               ^
irods-externals/json3.10.4-0/include/nlohmann/detail/input/input_adapters.hpp:129:7: note: candidate constructor (the implicit copy constructor) not viable: no known conversion from 'iterator_input_adapter<nlohmann::detail::iterator_input_adapter_factory<char *>::iterator_type>' to 'const iterator_input_adapter<const char *>' for 1st argument
class iterator_input_adapter
      ^
irods-externals/json3.10.4-0/include/nlohmann/detail/input/input_adapters.hpp:129:7: note: candidate constructor (the implicit move constructor) not viable: no known conversion from 'iterator_input_adapter<nlohmann::detail::iterator_input_adapter_factory<char *>::iterator_type>' to 'iterator_input_adapter<const char *>' for 1st argument
class iterator_input_adapter
      ^
irods-externals/json3.10.4-0/include/nlohmann/detail/input/input_adapters.hpp:134:5: note: candidate constructor not viable: requires 2 arguments, but 1 was provided
    iterator_input_adapter(IteratorType first, IteratorType last)
    ^
```

Without the braces, it uses `(2)`, which is the version the deprecation warning suggests:
https://github.com/nlohmann/json/blob/834918018eedf1b3737d87a9dfb7462db13c1a23/include/nlohmann/json.hpp#L6907